packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -2621,7 +2620,7 @@ class RenderTransform extends RenderProxyBox {
             oldLayer: layer is TransformLayer ? layer as TransformLayer? : null,
           );
         } else {
-          super.paint(context, offset + childOffset);
+          super.paint(context, offset);
```

Comment:
This doesn't seem right. It looks like you are dropping the translation here

Reply:
but that means you are unconditionally pushing a layer instead of adjusting the paint offset

Reply:
unless pushTransform is conditional

Reply:
(its not, it still does a saveLayer).

I'm concerned this is a bug elsewhere. Adjusting the paint offset should be fine

Reply:
I'm not really sure if this is the root cause. I didn't find any other way to fix that issue. It would be great if you could find a better solution.

There is also a similar issue here https://github.com/flutter/flutter/issues/111863 , I haven't carefully analyzed the cause of this issue, but the two issues are very similar.

Reply:
I think the root cause is that the ImageFiltered matrix in the example in the issue is a rotate around the origin and if you put a transform in then the origin is in a different location than if you just "move" the rendering to a new location.

Reply:
Another solution would be to only push the transform layer when we have a non-0 offset in an ImageFiltered widget? (Perhaps and only if the filter is a Matrix one?)

Reply:
I believe the image filter layer constructor assumes it can accept an offset by applying that to the matrix transformation. But if that isn't correct we could fix this by making the offset a separate property and handle that similarly to how we do so for the opacity layer